src/lua/serializer.c

```diff
@@ -76,12 +76,32 @@ static struct {
 	OPTION(LUA_TBOOLEAN, encode_use_tostring, 0),
 	OPTION(LUA_TBOOLEAN, encode_invalid_as_nil, 0),
 	OPTION(LUA_TBOOLEAN, encode_error_as_ext, 1),
+	OPTION(LUA_TBOOLEAN, encode_esc_slash, 0),
 	OPTION(LUA_TBOOLEAN, decode_invalid_numbers, 1),
 	OPTION(LUA_TBOOLEAN, decode_save_metatables, 1),
 	OPTION(LUA_TNUMBER,  decode_max_depth, 128),
 	{ NULL, 0, 0, 0},
 };
 
+/** toggler for json.encode behavior change with tarantool.compat (gh-7000) */
+int
+json_esc_slash_toggle(struct lua_State *L)
```

Comment:
I doubt that 'toogle' is a good term here: a toogle would allow to switch from 'off' to 'on' and vice verse. Here we have a usual setter.

Following our usual naming conventions we should name it, for example, `lbox_serializer_encode_forward_slash_default()`. `lbox_` is here, because it is supposed to be called from the Lua land. `serializer` is here instead of `json`, because, in fact, the function changes a default for a serializer option. Sure, this option affects only json encoder, but the list of options is owned by the serializer module.

However this name looks too wordy. I would rather implement the same for any option: `lbox_serializer_default_cfg()`. I remember that there was something of this kind in a previous version of the patch and I was against a complicated macro usage. I think that it is possible to generalize the implementation and don't miss readability.

Next, we usually expose `lbox_foo_bar` from module `foo` as field `bar`. I don't see a reason to go in another way here. I would add `internal.serializer.default_cfg()`, which accepts a table of option names as keys and its default values as values.

What do you think?

Reply:
Currently, we don't have `internal.serializer` module. I don't think this is the time or the PR to introduce it.

Reply:
Added `lbox_serializer_default_cfg()` and moved `json_escape_forward_slash_toggle()` to `compat.c`.

Reply:
We discussed naming with @Totktonada and I changed them accordingly

Reply:
Summarized my opinion in https://github.com/tarantool/tarantool/pull/6910#issuecomment-1260924239.